src/parameterizations/vertical/MOM_geothermal.F90

```diff
@@ -136,6 +144,10 @@ subroutine geothermal(h, tv, dt, ea, eb, G, GV, CS, halo)
 !$OMP                                  wt_in_place,dTemp,dRcv,h_transfer,heating, &
 !$OMP                                  I_h)
 
+! Save temperature and thickness before any changes are made (for diagnostic)
+temp_old = tv%T
+h_old = h
```

Comment:
`temp_old` and `h_old` should be hidden behind an `if (CS%id_internal_heat_tend_3d > 0) then`

Reply:
Thanks.